README.rst

```diff
@@ -0,0 +1,591 @@
+.. _async sandwich: https://trio.readthedocs.io/en/latest/tutorial.html#async-sandwich
+.. _always propagate: https://trio.readthedocs.io/en/latest/design.html#exceptions-always-propagate
+.. _causality: https://vorpus.org/blog/some-thoughts-on-asynchronous-api-design-in-a-post-asyncawait-world/#c-c-c-c-causality-breaker
+.. _shared nothing architecture: https://en.wikipedia.org/wiki/Shared-nothing_architecture
+.. _cancellation: https://trio.readthedocs.io/en/latest/reference-core.html#cancellation-and-timeouts
+.. _channels: https://en.wikipedia.org/wiki/Channel_(programming)
+
+
+Install
+-------
+No PyPi release yet!
+
+::
+
+    pip install git+git://github.com/tgoodlet/tractor.git
+
+
+What's this? Spawning event loops in subprocesses?
+--------------------------------------------------
+Close, but not quite.
+
+The first step to grok ``tractor`` is to get the basics of ``trio``
+down. A great place to start is the `trio docs`_ and this `blog post`_
+by njsmith_.
+
+``tractor`` takes much inspiration from pulsar_ and execnet_ but attempts to be much more
+minimal, focus on sophistication of the lower level distributed architecture,
+and of course does **not** use ``asyncio``, hence **no** event loops.
+
+.. _trio docs: https://trio.readthedocs.io/en/latest/
+.. _pulsar: http://quantmind.github.io/pulsar/design.html
+.. _execnet: https://codespeak.net/execnet/
+.. _blog post: https://vorpus.org/blog/notes-on-structured-concurrency-or-go-statement-considered-harmful/
+.. _njsmith: https://github.com/njsmith/
+
+
+A trynamic first scene
+----------------------
+As a first example let's spawn a couple *actors* and have them run their lines:
+
+.. code:: python
+
+    import tractor
+    from functools import partial
+
+    _this_module = __name__
+    the_line = 'Hi my name is {}'
+
+
+    async def hi():
+        return the_line.format(tractor.current_actor().name)
+
+
+    async def say_hello(other_actor):
+        await trio.sleep(0.4)  # wait for other actor to spawn
+        async with tractor.find_actor(other_actor) as portal:
+            return await portal.run(_this_module, 'hi')
+
+
+    async def main():
+        """Main tractor entry point, the "master" process (for now
+        acts as the "director").
+        """
+        async with tractor.open_nursery() as n:
+            print("Alright... Action!")
+
+            donny = await n.start_actor(
+                'donny',
+                main=partial(say_hello, 'gretchen'),
+                rpc_module_paths=[_this_module],
+                outlive_main=True
+            )
+            gretchen = await n.start_actor(
+                'gretchen',
+                main=partial(say_hello, 'donny'),
+                rpc_module_paths=[_this_module],
+            )
+            print(await gretchen.result())
+            print(await donny.result())
+            await donny.cancel_actor()
+            print("CUTTTT CUUTT CUT!!?! Donny!! You're supposed to say...")
```

Comment:
Might not want a question mark here. Why would the director be unsure if they need to call cut?